frame/system/src/lib.rs

```diff
@@ -566,8 +573,12 @@ decl_module! {
 		/// - `O(D)` where `D` length of `Digest`
 		/// - 1 storage write or delete (codec `O(1)`).
 		/// - 1 call to `deposit_log`: `O(D)` (which depends on the length of `Digest`)
+		/// - Base Weight: 9.29 + 0.511 * d µs
+		/// - DB Weight:
+		///     - Reads: System Digest
+		///     - Writes: Changes Trie, System Digest
 		/// # </weight>
-		#[weight = (20_000_000, DispatchClass::Operational)]
+		#[weight = (T::DbWeight::get().reads_writes(1, 2) + 10_000_000, DispatchClass::Operational)]
```

Comment:
This weight is wrong, we need to factor in D, the length of `Digest`.

It seems that this storage item is similar to events (a growing vec). We may want to apply the same change we made to System.Events here.

Reply:
I guess this must be addressed before merging?

Reply:
Does this need to be addressed before merging?

Reply:
It is fixed now! D is no longer a factor with the append api